scripts/ci/update_server_names.py

```diff
@@ -0,0 +1,156 @@
+        "cities": city_names,
+    }
+
+
+def pascalize(string):
+    output = ""
+    for chunk in string.split(" "):
+        output += chunk[0].upper()
+        output += chunk[1:]
+    return output
+
+
+def get_city_translation_id(str):
+    # Remove the state suffix e.g. São Paulo, SP -> São Paulo
+    str = str.split(",")[0]
+    # Remove any unexpected characters e.g. São Paulo -> SoPaulo
+    str = re.sub(r'[^a-zA-Z ]', '', str)
+
+    return pascalize(str)
+
+
+if __name__ == "__main__":
+    ###
+    # 1. Fetch the latest list of servers from Mullvad.
+
+    countries, cities = fetch_server_list().values()
+
+    # Create a map like so "<xliff_id>": "<xliff_source>"
+    string_map = {}
+    for country in countries:
+        string_map[country['code']] = country['name']
+    for city in cities:
+        id = get_city_translation_id(city)
+        string_map[id] = city
+
+    ###
+    # 2. Fetch the list of servers in extras.xliff
+
+    script_path = os.path.dirname(os.path.abspath(__file__))
+    xliff_path = os.path.join(
+        script_path, os.path.pardir, os.path.pardir, 'src', 'translations',
+        'extras', 'extras.xliff'
+    )
+    if not os.path.exists(xliff_path):
+        sys.exit(f"extras.xliff not found in path {xliff_path}")
+
+    tree = etree.parse(xliff_path)
+    root = tree.getroot()
+
+    found_server_names = []
+    for trans_unit in root.findall('.//{urn:oasis:names:tc:xliff:document:1.2}trans-unit'):
+        unit_id = trans_unit.get('id')
+        if unit_id.startswith('servers.'):
+            found_server_names.append(unit_id.split('.')[1])
+
+    ###
+    # 3. Get the list of missing servers strings and update extras.xliff
+
+    missing = list(set(string_map.keys()) - set(found_server_names))
+    # No missing strings, return early
+    if not missing:
+        print("Server list is up to date!")
+        sys.exit(0)
+
+    missing_string_map = { key: value for key, value in string_map.items() if key in missing}
+
+    # Iterate over the entries and add them to the XLIFF tree
+    servers_node = root.find(".//{urn:oasis:names:tc:xliff:document:1.2}*[@original='../src/apps/vpn/ui/screens/home/ViewServers.qml']")
+    if not servers_node:
+        sys.exit("Unable to find servers node. Has the extras.xliff file been changed?")
+
+    for id, source in missing_string_map.items():
+        new_unit = etree.SubElement(servers_node, "{urn:oasis:names:tc:xliff:document:1.2}trans-unit")
+        new_unit.set("id", f"servers.{id}")
+
+        new_unit.set("{urn:oasis:names:tc:xliff:document:1.2}xml:space", "preserve")
+
+        source_node = etree.SubElement(new_unit, "{urn:oasis:names:tc:xliff:document:1.2}source")
+        source_node.text = source
+
+    # Save the new tree to extras.xliff file
+    with open(xliff_path, "w", encoding="utf-8") as fp:
+        etree.register_namespace('',"urn:oasis:names:tc:xliff:document:1.2")
+        # Fix indentation
+        reindent(root)
```

Comment:
What happens if we don't reindent the file?

Reply:
The file formatting becomes completely different from what is the l10n repo.